sklearn/preprocessing/_data.py

```diff
@@ -1585,8 +1585,7 @@ def transform(self, X):
                         columns.append(bias)
                 XP = sparse.hstack(columns, dtype=X.dtype).tocsc()
             else:
-                XP = np.empty((n_samples, self.n_output_features_),
-                              dtype=X.dtype, order=self.order)
+                XP = np.empty_like(X, dtype=X.dtype, order=self.order)
```

Comment:
unfortunately `self.n_output_features_` isn't equal to `X.shape[1]`

(side note, `empty_like` also takes care of the dtype so you don't need to specify it)

Reply:
Yeah, I didn't run the sklearn test, but was hitting a different issue in vaex first. It should pass now. Indeed, dtype is not needed.